joml-geometry/src/main/java/org/terasology/joml/geom/Rectangleic.java

```diff
@@ -91,270 +97,240 @@
-     * If the two rectangles do not intersect, then the minimum coordinates of <code>dest</code>
-     * will have a value of {@link Integer#MAX_VALUE} and the maximum coordinates will have a value of
-     * {@link Integer#MIN_VALUE}.
-     *
-     * @param other
-     *          the other rectangle
-     * @param dest
-     *          will hold the result
+     * This is equivalent to <br>
+     * <pre>
+     *     translate(-ax, -ay);
+     *     scale(sx, sy);
+     *     translate(ax, ay);
+     * </pre>
+     *
+     * @param sx the scaling factor on the x axis
+     * @param sy the scaling factor on the y axis
+     * @param ax the x coordinate of the anchor
+     * @param ay the y coordinate of the anchor
+     * @param dest will hold the result
      * @return dest
      */
-    Rectanglei intersection(Rectangleic other, Rectanglei dest);
+    Rectanglei scale(int sx, int sy, int ax, int ay, Rectanglei dest);
+
+    /**
+     * Scale <code>this</code> about an anchor and store the result in <code>dest</code>.
+     * <p>
+     * This is equivalent to <br>
+     * <pre>
+     *     translate(anchor.negate());
+     *     scale(sx, sy);
+     *     translate(anchor.negate());
+     * </pre>
+     *
+     * @param sx the scaling factor on the x axis
+     * @param sy the scaling factor on the y axis
+     * @param anchor the location of the anchor
+     * @param dest will hold the result
+     * @return dest
+     */
+    Rectanglei scale(int sx, int sy, Vector2ic anchor, Rectanglei dest);
 
     /**
      * Check if this rectangle contains the given <code>rectangle</code>.
      *
-     * @param rectangle
-     *          the rectangle to test
+     * @param rectangle the rectangle to test
      * @return <code>true</code> iff this rectangle contains the rectangle; <code>false</code> otherwise
      */
     boolean containsRectangle(Rectangledc rectangle);
 
-
     /**
      * Check if this rectangle contains the given <code>rectangle</code>.
      *
-     * @param rectangle
-     *          the rectangle to test
+     * @param rectangle the rectangle to test
      * @return <code>true</code> iff this rectangle contains the rectangle; <code>false</code> otherwise
      */
     boolean containsRectangle(Rectanglefc rectangle);
 
     /**
      * Check if this rectangle contains the given <code>rectangle</code>.
      *
-     * @param rectangle
-     *          the rectangle to test
+     * @param rectangle the rectangle to test
      * @return <code>true</code> iff this rectangle contains the rectangle; <code>false</code> otherwise
      */
     boolean containsRectangle(Rectangleic rectangle);
 
     /**
      * Check if this rectangle contains the given <code>point</code>.
      *
-     * @param point
-     *          the point to test
+     * @param point the point to test
      * @return <code>true</code> iff this rectangle contains the point; <code>false</code> otherwise
      */
     boolean containsPoint(Vector2ic point);
```

Comment:
I'm kind of hesitant about this function. might be a bit better to only accepting the floating point variations. did we decided if the edge of a Rectanglei is inclusive.  to an extent the intersects is in a similar situation. is a rectangle intersecting if its only the edge? wonder if it would be better to just leave it up to the person using this rather then proving some concrete method.

Reply:
That's exactly the reasoning why this moved out of JOML and into our org. I think we should define the behavior to be used for Terasology so that we don't have any confusion about it throughout the code base.

Reply:
This is used in `BrowserWidget`, `DistrictOverlay` and `MinimapIconOverlay`.

Reply:
Thinking about properties that should hold:

```java
assertTrue(rect.contains(rect)); // rectangle should contain itself
```

Now I'm wondering which parts of the rectangle are inclusive and which are exclusive. I don't think we have to think about the edges on it's own, but only about the min and max corner:

```java
assertTrue(rect.contains(rect.getMin(new Vector2i()));
assertTrue(rect.contains(rect.getMax(new Vector2i()));
// currently both hold	
```

If both are true, then there's an overlap/intersection between two rectangles right next to each other. If only one is, you can "tile" touching rectangles without intersection. 

```java
Rectanglei rectA = new Rectanglei(0, 0, 1, 1);
Rectanglei rectB = new Rectanglei(0, 1, 1, 2);

assertTrue(rectA.intersectsRectangle(rectB));
assertEquals(new Rectanglei(0, 1, 1, 1), rectA.intersection(rectB, new Rectanglei());
```

Reply:
Started https://github.com/MovingBlocks/joml-ext/pull/23 for writing tests on `Rectanglei` to verify whatever our contract is going to be. Also gives an insight on how the current implementation behaves...